app/models/pawn.rb

```diff
@@ -1,2 +1,42 @@
 class Pawn < Piece
+  def valid_pawn_move?(to_x:, to_y:)
+    return true if direction_of_white_and_black_pawns(to_y) &&
+      (
+        normal_move?(to_x, to_y) ||
+        first_move?(to_x, to_y) ||
+        pawn_diagonal?(to_x, to_y)
+      )
+    false
+  end
+
+  private
+
+  # This logic makes sure that the pawn can only move forward
+  # White pawns can only move *up* the board
+  # Black pawns can only move *down* the board
+  def direction_of_white_and_black_pawns(to_y)
+    return true if color == 'white' && to_y > y_position
+    return true if color == 'black' && to_y < y_position
+    false
+  end
+
+  # Returns true if the pawn advances one space from its origin
```

Comment:
This comment needs to be more specific, e.g: "Returns true if the pawn has already made its first move **and** advances one square from its current location."